repairnator/repairnator-scanner/src/main/java/fr/inria/spirals/repairnator/scanner/ProjectScanner.java

```diff
@@ -35,15 +35,31 @@
 
     private Collection<String> slugs;
     private Collection<Repository> repositories;
-    private Date limitDate;
 
-    private Date beginDate;
-    private Date endDate;
     private int lookupHours;
+    private Date lookFromDate;
+    private Date lookToDate;
     private LauncherMode launcherMode;
 
+    private Date scannerRunningBeginDate;
+    private Date scannerRunningEndDate;
+
     public ProjectScanner(int lookupHours, LauncherMode launcherMode) {
+        this(launcherMode);
         this.lookupHours = lookupHours;
+        Calendar limitCal = Calendar.getInstance();
+        limitCal.add(Calendar.HOUR_OF_DAY, -this.lookupHours);
+        this.lookFromDate = limitCal.getTime();
+        this.lookToDate = new Date();
+    }
+
+    public ProjectScanner(Date lookFromDate, Date lookToDate, LauncherMode launcherMode) {
+        this(launcherMode);
```

Comment:
You should define completely this one (without calling this), and use it in the other one. The constructor with only `LauncherMode` should not be authorized as it will throw an exception if you try to get builds from it.